cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java

```diff
@@ -94,17 +94,26 @@ public static void runProgram(Path sourceRootPath, Path sourcePath, String funct
         System.setProperty(ProjectDirConstants.BALLERINA_SOURCE_ROOT, fullPath.getParent().toString());
         loadConfigurations(fullPath.getParent(), runtimeParams, configFilePath, observeFlag);
 
+        boolean isProject = RepoUtils.hasProjectRepo(sourceRootPath);
         if (srcPathStr.endsWith(BLANG_EXEC_FILE_SUFFIX)) {
             programFile = BLangProgramLoader.read(sourcePath);
-        } else if (Files.isRegularFile(fullPath) &&
-                srcPathStr.endsWith(BLANG_SRC_FILE_SUFFIX) &&
-                !RepoUtils.hasProjectRepo(sourceRootPath)) {
+        } else if (Files.isRegularFile(fullPath) && srcPathStr.endsWith(BLANG_SRC_FILE_SUFFIX) && !isProject) {
             programFile = compile(fullPath.getParent(), fullPath.getFileName(), offline);
         } else if (Files.isDirectory(sourceRootPath)) {
-            if (Files.isDirectory(fullPath) && !RepoUtils.hasProjectRepo(sourceRootPath)) {
+            // If we are trying to run a package from outside a project
+            if (Files.isDirectory(fullPath) && !isProject) {
                 throw LauncherUtils.createLauncherException(
-                        "error: did you mean to run the Ballerina package as a project? If so run 'ballerina init' to" +
-                                " make it a project with a .ballerina directory");
+                        "error: did you mean to run the Ballerina package as a project? If so either run from within " +
+                                "the project or use --sourceroot to specify the project and run the package");
```

Comment:
did you mean to run the package ? If so, either run from the project folder or use --sourceroot to specify the project path and run the package.